src/com/company/Main.java

```diff
@@ -12,14 +12,14 @@ public static void main(String[] args){
 
         String txtfile = args[0];
         File file = new File(txtfile);
-        checkforsupport(file);
-        checkforexistance(file);
+        checkForSupport(file);
+        checkForExistance(file);
         String searchword = args[1];
         System.out.println("processing................");
         String data = null;
         data = readFileAsString(txtfile);
```

Comment:
What if the file is empty? If we do a check there, we can stop the execution then and there and no need to proceed further.

Reply:
if file is empty the message is passed like the file has no data and execution will be terminated